agent/agent.go

```diff
@@ -205,6 +205,9 @@ func (a *Agent) run(ctx context.Context) {
 		sessionq      chan sessionOperation
 		leaving       = a.leaving
 		subscriptions = map[string]context.CancelFunc{}
+		// subscriptionDone is a channel that allows us to notify ourselves
+		// that a lot subscription should be finished.
+		subscriptionDone = make(chan string, 50)
```

Comment:
What's the reasoning for having this buffered? Why 50 and not 1 or 0?

I think no matter what we choose there is potential for goroutines to pile up waiting for the send.

Reply:
I've changed the channel to unbuffered. You're right, buffering doesn't help anything.